### ISSUE TITLE
Ceph version detect crash loop

### ISSUE THREAD

**Issue by JiangHuPZ** [rook/rook#14595]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Hi team, recently, we noticed an unexpected problem with starting rook-ceph on some customized systems: The operator cannot check the version of ceph at the beginning running stage. And after that, it goes to a version detect crash loop.

![image](https://github.com/user-attachments/assets/20047146-369c-4855-8446-3d1b2c1c5dfd)
![image](https://github.com/user-attachments/assets/28a5e95d-4a46-4163-bf1c-9554bc272bb2)
![image](https://github.com/user-attachments/assets/db03b556-679b-4a43-8b75-1832ceebcc02)

This is leading to failure in reconciliation of CephObjectStore, CephObjectStoreUser and CephCluster. And the rook-ceph-tools also cannot create.

![image](https://github.com/user-attachments/assets/cfe3e25b-8f9b-4393-b880-c65bfb92ce30)

However we can successfully run the ceph version check manually
```shell
$ ctr -n k8s.io run --rm --tty ceph:v18.2.2 ceph-version-test ceph --version
ceph version 18.2.2 (531c0d11a1c5d39fbfe6aa8a521f023abf3bf3e2) reef (stable)
```

**Environment**:
* CRI: containerd
* OS (e.g. from /etc/os-release): alios or kylin
* Kernel (e.g. `uname -a`): 4.19.91
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): v1.13.10
* Storage backend version (e.g. for ceph do `ceph -v`): v18.2.2
* Kubernetes version (use `kubectl version`): 1.26.15
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): baremetal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

**Comment by travisn** [comment]
@JiangHuPZ Do the errors look the same as #14397?

**Comment by JiangHuPZ** [comment]
> @JiangHuPZ Do the errors look the same as #14397?

Thanks for reply.

This is indeed a very similar problem, but after we checked the operator.yaml, we confirmed that resource limit related fields are declared in it.
```yaml
resources:
  limits:
   cpu: 500m
   memory: 512Mi
  requests:
   cpu: 100m
   memory: 128Mi
```

But we can start the rook ceph cluster normally with the same configuration file  on other systems(CentOS 7.9, AnolisOS 8.8).

Is it possible that some hidden system setting triggers this issue and how can we troubleshoot this?
